signale.js

```diff
@@ -163,7 +163,7 @@ class Signale {
         [msg] = args;
       } else {
         let [{prefix, message, suffix}] = args;
-        if(message === undefined) {
+        if (message === undefined) {
```

Comment:
I am doing like this **>>** `msg = message ? util.format(...message) : util.format(...args);`

Reply:
Hmm, if `message` is undefined, which means that we have no message to log, the `args` object will at best only hold the `suffix` and `prefix` attributes, thus they will be printed twice, as they are handled independently on lines [175](https://github.com/klauscfhq/signale/blob/master/signale.js#L174) & [212](https://github.com/klauscfhq/signale/blob/master/signale.js#L212), for example:

```js
// running with: `msg = message ? util.format(...message) : util.format(...args);`
const signale = require('signale');

signale.complete({prefix: '[task]', suffix: '(@santa)'}); // `message` is undefined
// => [task] ☒  complete  { prefix: '[task]', suffix: '(@santa)' } (@santa)

// The printed object `{ prefix: '[task]', suffix: '(@santa)' }` is unexpected, since 
// we left `message` undefined, which means nothing should have been printed in its place
```

```js
// running with: `msg = message ? util.format(...message) : '';`
const signale = require('signale');

signale.complete({prefix: '[task]', suffix: '(@santa)'}); // `message` is undefined
// => [task] ☒  complete   (@santa)

// `message` was not defined and as it was explected nothing was printed in its the place  
```

Reply:
I edited the markdown of the above comment so the example can look better and be more clear. Please let me know what you think about this approach : )

Reply:
If i do like `msg = message ? util.format(...message) : '';` then it will not print message for below example:
```js
signale.success({a: 2});
signale.success([1, 2]);
````

Reply:
Somehow I managed previous issue occurrence. By like this

```js
// for fix xo warning
_prefixSufixNestedTernary(prefix, suffix, args) {
    return prefix || suffix ? '' : util.format(args);
}

msg = message ? util.format(...message) : this._prefixSufixNestedTernary(prefix, suffix, ...args);
```
Is this fine?

Reply:
Yep, you are totally right, that is an issue also described in #24. At this moment signale loggers can only accept one or more comma-delimited strings or an object containing the `suffix`, `prefix` and `message` attributes, which means that currently the only way to print any other object is to use **template literals**, which on their own are not such a perfect replacement either: 

```js
const object = {a: 2};
const array = [1, 2];

signale.success(`${object}`);
//=> ✔  success   [object Object]
signale.success(`${array}`);
//=> ✔  success   1,2
```

But with string interpolation it will be possible to cover these cases effectively:

```js
signale.success('%O', {a: 2});
//=> ✔  success   { a: 2 }
signale.success('%O', [1, 2]);
//=> ✔  success   [ 1, 2 ]
```

Thus it is acceptable to use `msg = message ? util.format(...message) : '';` in order to deliver string interpolation and also to preserve the fuctionality of signale when using the `{suffix, prefix, message}` object where the `message` attribute can be `undefined`. The idial case is exactly [what you described](https://github.com/klauscfhq/signale/pull/23#discussion_r194232395), and it is something will be added as soon as possible, as a feature on its own, since it will be possible to log any object directly, in the same fashion that `console.log` does : )

Reply:
I hope except `signale.success(`${object}`);` case other will works fine.